### ISSUE TITLE
[RLLib] Algorithm Reset_Config() method for Reusing Actors

### ISSUE THREAD

**Issue by dkupsh** [ray-project/ray#38136]
### Description

Adding the Reset_Config() method to the Algorithm class.

Something similar to this:
`
def reset_config(self, new_config : dict) -> Bool: 
        self.config = self.get_default_config().update_from_dict(new_config)
        return True
`
However, when I tried implementing this, there were future errors that arose.

### Use case

This would be helpful for integrating Tune more with RLLIB. It would allow for larger actors to run without needing to be recreated during hyperparameter, using the Reuse_Actors option.

**Comment by sven1977** [comment]
Yeah, it would be really cool to have this. However, we need to be very careful, b/c some settings in the config are NOT changeable because they influence the model-architecture and thus would break checkpoint recovery.
Maybe we can validate the `new_config` for having these changes or not and if not proceed with updating the config inside the running algo.

**Comment by dkupsh** [comment]
The problem I had when implementing the change was that the old config was a dictionary (due to tune.fit) and so:

```
def reset_config(self, new_config : dict) -> Bool: 
       return True
```

would give an error due to it being a dictionary. 

Even doing:
```
def reset_config(self, new_config : dict) -> Bool: 
       self.config = self.get_default_config().update_from_dict(self.config)
       return True
```
gave errors when running.

**Comment by PhilippWillms** [comment]
@sven1977 : Stumbling today across this issue, maybe this is something which can be improved with new RLlib API stack ?
